### ISSUE TITLE
INTERNALERROR> SystemError: AST constructor recursion depth mismatch (before=78, after=133)

### ISSUE THREAD

**Issue by coneybeare** [pytest-dev/pytest#11724]
# Summary

We are seeing a very weird error in which coverage runs with html outputs are raising this `INTERNALERROR`

executed command and options:

```
pytest --cov app/ --cov-report term-missing --cov-report html:coverage --cov-branch --cov-append -vv
```

Error traceback
```
INTERNALERROR> Traceback (most recent call last):00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/_pytest/main.py", line 270, in wrap_session00:14
INTERNALERROR>     session.exitstatus = doit(config, session) or 000:14
INTERNALERROR>                          ^^^^^^^^^^^^^^^^^^^^^00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/_pytest/main.py", line 324, in _main00:14
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/pluggy/_hooks.py", line 433, in __call__00:14
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)00:14
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/pluggy/_manager.py", line 112, in _hookexec00:14
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)00:14
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/pluggy/_callers.py", line 133, in _multicall00:14
INTERNALERROR>     teardown[0].send(outcome)00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/pytest_cov/plugin.py", line 307, in pytest_runtestloop00:14
INTERNALERROR>     self.cov_total = self.cov_controller.summary(self.cov_report)00:14
INTERNALERROR>                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/pytest_cov/engine.py", line 44, in ensure_topdir_wrapper00:14
INTERNALERROR>     return meth(self, *args, **kwargs)00:14
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/pytest_cov/engine.py", line 189, in summary00:14
INTERNALERROR>     total = self.cov.html_report(ignore_errors=True, directory=output)00:14
INTERNALERROR>             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/coverage/control.py", line 1157, in html_report00:14
INTERNALERROR>     ret = reporter.report(morfs)00:14
INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/coverage/html.py", line 313, in report00:14
INTERNALERROR>     self.write_html_file(ftr, prev_html, next_html)00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/coverage/html.py", line 388, in write_html_file00:14
 00:14
INTERNALERROR>                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/coverage/html.py", line 130, in data_for_file00:14
INTERNALERROR>     for lineno, tokens in enumerate(fr.source_token_lines(), start=1):00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/coverage/phystokens.py", line 120, in source_token_lines00:14
INTERNALERROR>     match_case_lines = MatchCaseFinder(source).match_case_lines00:14
INTERNALERROR>                        ^^^^^^^^^^^^^^^^^^^^^^^00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/site-packages/coverage/phystokens.py", line 85, in __init__00:14
INTERNALERROR>     self.visit(ast.parse(source))00:14
INTERNALERROR>                ^^^^^^^^^^^^^^^^^00:14
INTERNALERROR>   File "/usr/local/lib/python3.11/ast.py", line 50, in parse00:14
INTERNALERROR>     return compile(source, filename, mode, flags,00:14
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^00:14
INTERNALERROR> SystemError: AST constructor recursion depth mismatch (before=78, after=133)
```

We also cannot reproduce locally, only on CI (semaphore), very strange.

## Expected vs actual result

I would expect the coverage not to raise this error. However I am not entirely convinced it is a bug, and not a config issue. Please educate me.

# Reproducer

## Versions

Output of relevant packages `pip list`, `python --version`, `pytest --version` etc.

```
pytest==7.4.0
pytest-cov==4.1.0
```

```
% python --version
Python 3.11.3
```

Make sure you include complete output of `tox` if you use it (it will show versions of various things).

## Config

Include your `tox.ini`, `pytest.ini`, `.coveragerc`, `setup.cfg` or any relevant configuration.

## Code

Link to your repository, gist, pastebin or just paste raw code that illustrates the issue.

If you paste raw code make sure you quote it, eg:

```python
def foobar():
    pass
```

**Comment by brianzi** [comment]
I encountered this problem today and I'm pretty sure the cause is a bug in the parser of cpython itself (https://github.com/python/cpython/issues/106905), which has been around in that form in python3.11 releases since 3.11.0b4 up to and including 3.11.7.

But this bug has now been fixed though, and indeed, for me, switching from 3.11.7 to a post-3.11.7 build has removed the issue. That bugfix is also listed already in the [announcement for the next 3.11 release](https://docs.python.org/3.11/whatsnew/changelog.html#python-next)

Hope this helps!
